WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteSourceManager.kt

```diff
@@ -0,0 +1,123 @@
+import org.wordpress.android.ui.mysite.dynamiccards.DynamicCardsSource
+import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
+import org.wordpress.android.util.config.MySiteDashboardPhase2FeatureConfig
+import javax.inject.Inject
+
+@Suppress("LongParameterList")
+class MySiteSourceManager @Inject constructor(
+    private val analyticsTrackerWrapper: AnalyticsTrackerWrapper,
+    private val currentAvatarSource: CurrentAvatarSource,
+    private val domainRegistrationSource: DomainRegistrationSource,
+    private val dynamicCardsSource: DynamicCardsSource,
+    private val quickStartCardSource: QuickStartCardSource,
+    private val scanAndBackupSource: ScanAndBackupSource,
+    private val selectedSiteSource: SelectedSiteSource,
+    postCardsSource: PostCardsSource,
+    siteIconProgressSource: SiteIconProgressSource,
+    private val mySiteDashboardPhase2FeatureConfig: MySiteDashboardPhase2FeatureConfig
+) {
+    private val mySiteSources: List<MySiteSource<*>> = listOf(
+            selectedSiteSource,
+            siteIconProgressSource,
+            quickStartCardSource,
+            currentAvatarSource,
+            domainRegistrationSource,
+            scanAndBackupSource,
+            dynamicCardsSource,
+            postCardsSource
+    )
+
+    fun build(siteLocalId: Int?, coroutineScope: CoroutineScope): List<LiveData<out PartialState>> {
+        return if (siteLocalId != null) {
+            mySiteSources.map { source -> source.build(coroutineScope, siteLocalId).distinctUntilChanged() }
+        } else {
+            mySiteSources.filterIsInstance(SiteIndependentSource::class.java)
+                    .map { source -> source.build(coroutineScope).distinctUntilChanged() }
+        }
+    }
+
+    fun isRefreshing(): Boolean {
+        if (mySiteDashboardPhase2FeatureConfig.isEnabled()) {
+            mySiteSources.filterIsInstance(MySiteRefreshSource::class.java).forEach {
+                if (it.isRefreshing() == true) {
+                    return true
+                }
+            }
+        }
+        return false
+    }
+
+    fun refresh() {
+        if (mySiteDashboardPhase2FeatureConfig.isEnabled()) {
+            refreshAllSources()
+        } else {
+            refreshSelectSources()
+        }
+    }
+
+    fun onResume(isFirstResume: Boolean) {
+        when (isFirstResume) {
+            true -> refreshSelectSources()
+            false -> if (mySiteDashboardPhase2FeatureConfig.isEnabled()) {
+                refreshAllSources()
+            } else {
+                refreshSelectSources()
+            }
+        }
+    }
+
+    fun clear() {
+        domainRegistrationSource.clear()
+        scanAndBackupSource.clear()
+        selectedSiteSource.clear()
+    }
+
+    private fun refreshAllSources() {
+        mySiteSources.filterIsInstance(MySiteRefreshSource::class.java).forEach { it.refresh() }
+    }
+
+    private fun refreshSelectSources() {
```

Comment:
Question (❓): Does `select` sources mean something, like it is sources that have to do with some kind of selection? Or did you mean something like `selected` sources, meaning a subset of `all` sources, the selected sources?

Reply:
I used `select` as an adjective to describe the sources - meaning that only certain pre-determined sources would be refreshed. I didn't like "old" vs. "new" because even "old" wasn't quite correct and eventually "new" becomes old. I didn't use `selected` because that could imply that the user selected them, when in this case, the user didn't select anything. Naming is hard. 

I do like `refreshSubsetOfAllSources` or `refreshChosenSources` or `refreshPredeterminedSources` anything else you can think of 🙏

Reply:
Cool, thanks for the explanation Annmarie! 🌟 

I vote for `refreshSubsetOfAllSources`, I like that! 👍 Or `refreshSomeSources`... 🤔

Reply:
Going with `refreshSubsetOfAllSources` !!